lib/autoproj/package_managers/apt_dpkg_manager.rb

```diff
@@ -8,23 +8,28 @@ class AptDpkgManager < ShellScriptManager
             def initialize(ws, status_file = "/var/lib/dpkg/status")
                 @status_file = status_file
                 @installed_packages = nil
+                @installed_versions = {}
                 super(ws, true,
                       %w{apt-get install},
                       %w{DEBIAN_FRONTEND=noninteractive apt-get install -y})
             end
 
-            def self.parse_package_status(installed_packages, paragraph)
+            def self.parse_package_status(installed_packages, paragraph, installed_versions: {})
                 if paragraph =~ /^Status: install ok installed$/
                     if paragraph =~ /^Package: (.*)$/
-                        installed_packages << $1
+                        package_name = $1
+                        installed_packages << package_name
+                        if paragraph =~ /^Version: (.*)$/
+                            installed_versions[package_name] = $1
+                        end
                     end
                     if paragraph =~ /^Provides: (.*)$/
                         installed_packages.merge($1.split(',').map(&:strip))
                     end
                 end
             end
 
-            def self.parse_dpkg_status(status_file)
+            def self.parse_dpkg_status(status_file, installed_versions: {})
```

Comment:
Don't add a out argument like this when the method returns its parsed data. Just return both values. I'm guessing that you do this for backward compatibility, but these are meant to be internal APIs.

Reply:
Yep, that was the reason. Will refactor if you don't care about that